tests/suite.bats

```diff
@@ -409,3 +409,230 @@ EOF
+
+EOF
+  }
+
+  good_config2() {
+    cat <<EOF
+namespace: ${STACKER_NAMESPACE}
+stacks:
+  - name: update-rollback
+    class_path: stacker.tests.fixtures.mock_blueprints.Dummy2
+
+EOF
+  }
+
+  teardown() {
+    stacker destroy --force <(good_config)
+  }
+
+  stacker destroy --force <(good_config)
+
+  # Create the initial stack
+  stacker build <(good_config)
+  assert "$status" -eq 0
+  assert_has_line "Using default AWS provider mode"
+  assert_has_line "${STACKER_NAMESPACE}-update-rollback: pending"
+  assert_has_line "${STACKER_NAMESPACE}-update-rollback: submitted (creating new stack)"
+  assert_has_line "${STACKER_NAMESPACE}-update-rollback: complete (creating new stack)"
+
+  # Do a bad update and watch the rollback
+  stacker build <(bad_config)
+  assert "$status" -eq 1
+  assert_has_line "Using default AWS provider mode"
+  assert_has_line "${STACKER_NAMESPACE}-update-rollback: pending"
+  assert_has_line "${STACKER_NAMESPACE}-update-rollback: submitted (updating existing stack)"
+  assert_has_line "${STACKER_NAMESPACE}-update-rollback: submitted (rolling back update)"
+  assert_has_line "${STACKER_NAMESPACE}-update-rollback: failed (rolled back update)"
+
+  # Do a good update so we know we've correctly waited for rollback
+  stacker build <(good_config2)
+  assert "$status" -eq 0
+  assert_has_line "Using default AWS provider mode"
+  assert_has_line "${STACKER_NAMESPACE}-update-rollback: pending"
+  assert_has_line "${STACKER_NAMESPACE}-update-rollback: submitted (updating existing stack)"
+  assert_has_line "${STACKER_NAMESPACE}-update-rollback: complete (updating existing stack)"
+}
+
+
+@test "stacker build - handle rollbacks in dependent stacks" {
+  needs_aws
+
+  config() {
+    cat <<EOF
+namespace: ${STACKER_NAMESPACE}
+stacks:
+  - name: dependent-rollback-parent
+    class_path: stacker.tests.fixtures.mock_blueprints.Broken
+
+  - name: dependent-rollback-child
+    class_path: stacker.tests.fixtures.mock_blueprints.Dummy
+    requires: [dependent-rollback-parent]
+
+EOF
+  }
+
+  teardown() {
+    stacker destroy --force <(config)
+  }
+
+  stacker destroy --force <(config)
+
+  # Verify both stacks fail during creation
+  stacker build <(config)
+  assert "$status" -eq 1
+  assert_has_line "Using default AWS provider mode"
+  assert_has_line "${STACKER_NAMESPACE}-dependent-rollback-parent: pending"
+  assert_has_line "${STACKER_NAMESPACE}-dependent-rollback-child:  pending"
+  assert_has_line "${STACKER_NAMESPACE}-dependent-rollback-parent: submitted (creating new stack)"
+  assert_has_line "${STACKER_NAMESPACE}-dependent-rollback-parent: submitted (rolling back new stack)"
+  assert_has_line "${STACKER_NAMESPACE}-dependent-rollback-parent: failed (rolled back new stack)"
+  assert_has_line "${STACKER_NAMESPACE}-dependent-rollback-child:  failed (dependency has failed)"
+  assert_has_line "The following stacks failed their plans: dependent-rollback-parent, dependent-rollback-child"
```

Comment:
Minor nit. The noun "Plan" has always referred to the entire graph of stacks, so it's confusing for me to see it applied as a noun for an individual stack.

Maybe just:

> The following stacks failed: dependent-rollback-parent, dependent-rollback-child

Reply:
Makes sense and is actually shorter, thanks!

Reply:
Done.